File: src/frc/robot/modules/ISwerveModule.java
package frc.robot.modules;
import edu.wpi.first.math.geometry.Rotation2d;
import edu.wpi.first.math.kinematics.SwerveModuleState;

public interface ISwerveModule {
    public double getDrive();
    public Rotation2d getTurn();
    public SwerveModuleState getState();
    public void setDesiredState(SwerveModuleState desiredState, boolean preventTurn, boolean slow, boolean pid);
    public void simulationPeriodic(double dt);
    public void resetEncoders();
    public void setBrakeMode(boolean on);
    public void test();
}

File: src/main/kotlin/frc/robot/modules/ISwerveModule.kt
package frc.robot.modules;
import edu.wpi.first.math.geometry.Rotation2d;
import edu.wpi.first.math.kinematics.SwerveModuleState;
import frc.robot.modules.ModuleInfo;
interface ISwerveModule {
    fun getDrive(): Double;
    fun getTurn(): Rotation2d;
    fun getState(): SwerveModuleState;
    fun setDesiredState(desiredState: SwerveModuleState, preventTurn: Boolean, slow: Boolean, pid: Boolean);
    fun simulationPeriodic(dt: Double);
    fun resetEncoders();
    fun setBrakeMode(on: Boolean);
    fun test();
}